All_home_works/hw2/task4.py

```diff
@@ -0,0 +1,35 @@
+"""
+Write a function that accepts another function as an argument. Then it
+should return such a function, so the every call to initial one
+should be cached.
+
+
+def func(a, b):
+    return (a ** b) ** 2
+
+
+cache_func = cache(func)
+
+some = 100, 200
+
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+
+assert val_1 is val_2
+
+"""
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    memo = {}
+
+    def wrapper(*args, **kwargs):
+        call = str(args) + str(kwargs)
```

Comment:
Why do you need it?

Reply:
Because immutable keys are needed in a dictionary. And str is immutable.

Reply:
I am ok with this limitation if you will name the case when this approach will provide the incorrect result.

Reply:
At first, we will call the function with args, then we will call with kwargs. Example:
   cache(a=1) it will calculate  this;
   cache(1) it will calculate this also, and not take from the cache.

Reply:
Let's find another example of when this approach failes.